docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+Manager for each WDS. Currently this controller-manager runs in the KubeFlex hosting cluster.
+More details on the internals of this module are provided in [KubeStellar Controllers Architecture](#kubestellar-controllers-architecture)
+
+## Space Manager
+
+The Space Manager handles the lifecycle of spaces. 
+KubeStellar uses the [KubeFlex project](https://github.com/kubestellar/kubeflex)
+for space management. In KubeFlex, a space is named a `ControlPlane`, and we will use 
+both terms in this document. KubeSteller currently prereqs KubeFlex to 
+provide one or more spaces. We plan to make this optional in the near future.
+
+KubeFlex is a flexible framework that supports various kinds of control planes, such as:
+
+- *k8s*: a basic Kubernetes API Server with a subset of kube controllers. 
+This control plane does not run workloads (e.g., pods), and 
+it uses about 350 MB of memory per instance with a shared Postgres Database Backend.
+- *vcluster*: a virtual cluster that runs on the hosting cluster, 
+based on the  [vCluster Project](https://www.vcluster.com). This type control plane can run pods using worker nodes of the hosting cluster.
+- *host*: the KubeFlex hosting cluster, which is exposed as a control plane.
+- *external*: an external cluster that is imported as a control plane.
+- *ocm*: a control plane that uses the 
+[multicluster-controlplane project](https://github.com/open-cluster-management-io/multicluster-controlplane) 
+for managing multiple clusters.
+
+There are currently two roles for spaces managed by KubeFlex: Inventory and Transport Space 
+(ITS) and Workload Description Space (WDS). The former runs the [OCM Cluster Manager](#ocm-cluster-manager)
+on a vcluster-type control plane, and the latter runs on a k8s-type control plane.
+
+An ITS holds the OCM inventory (`ManagedCluster`) objects and mailbox namespaces. The mailbox 
+namespaces and their contents are implementation details that users do not deal with. Each 
+mailbox workspace corresponds 1:1 with a WEC and holds `ManifestWork` objects managed by the 
+central KubeStellar controllers.
+
+A WDS holds user workload objects and the user's objects that form the interface to KubeStellar 
+control. Currently the only control objects are `Placement` objects. We plan to add `Customization`
+objects soon, and later objects to specify summarization.
+
+When using KubeStellar with KubeFlex, users interact with the API server 
+of the hosting cluster to create or delete control planes, and with the 
+hosted API servers to manage *Workloads* and *Placements* or *ManagedClusters*.
+KubeFlex defines a ControlPlane CRD that represents a Control Plane. When a user
+initiates the creation of a Control Plane Custom Resource (CR) by
+executing the `kflex create <cp>` command or the `kubectl apply -f mycontrolplane.yaml` 
+command, the KubeFlex controller creates a new namespace within the hosting cluster, 
+and then deploys the following
+artifacts in that namespace:
+
```

Comment:
This section is an example to something that maybe should be explained in KF docs and just pointed from here

Reply:
Moved out to https://github.com/kubestellar/kubeflex/blob/main/docs/architecture.md and pointed to that section for more info.